pkg/key/key.go

```diff
@@ -0,0 +1,9 @@
+package key
+
+const (
+	CalicoVersionConstraint                       = ">= 3.10.0 < 3.15.0"
+	EtcdVersionConstraint                         = ">= 3.4.0 < 3.5.0"
+	KubernetesVersionConstraint                   = ">= 1.16.0 < 1.18.0"
```

Comment:
Why strictly lower than 1.18?
We already kind of successfully tested it. IMO most constraints should only enforce a minimum version - we want to test newer versions as easily as possible.

Reply:
My thinking is that we want to constrain versions to those that we have tested and already ensured that there are no errors. We found 3 warnings in component logs when testing 1.18 which I plan to fix today and release with wider version constraints. Worth discussing how we expect this to work though--changing constraints will require releasing a new operator which is extra work.

Reply:
I get your thought process - I just want to prevent the extra effort, when our team will be the ones raising these versions most of the time.
Limiting here makes it harder for us to try out new versions which I want to prevent at all cost :D

Reply:
I agree. It will limit our ability to automatically or manually test new Kubernetes minors. I would like to keep the maximum version constraint for etcd and calico as it will force us to think about possible issues and test first, but I don't mind dropping the maximum for Kubernetes here.

Reply:
Done.